source/client/stream_decoder.cc

```diff
@@ -87,11 +87,15 @@ void StreamDecoder::onPoolReady(Envoy::Http::StreamEncoder& encoder,
     // we take the risk of erroneously reporting that we did send all the bytes, instead of always
     // reporting 0 bytes.
     stream_info_.addBytesReceived(request_body_size_);
-    // TODO(oschaaf): We can probably get away without allocating/copying here if we pregenerate
-    // potential request-body candidates up front. Revisit this when we have non-uniform request
-    // distributions and on-the-fly reconfiguration in place.
-    std::string body(request_body_size_, 'a');
-    Envoy::Buffer::OwnedImpl body_buffer(body);
+    // Revisit this when we have non-uniform request distributions and on-the-fly reconfiguration in
+    // place. The string size below MUST match the cap we put on RequestOptions::request_body_size
+    // in api/client/options.proto!
+    static const std::string body(4194304, 'a');
```

Comment:
Due to static initialization fiasco, I recommend we follow Envoy practice around `CONSTRUCT_ON_FIRST_USE`

Reply:
> Due to static initialization fiasco, I recommend we follow Envoy practice around `CONSTRUCT_ON_FIRST_USE`

I have a surprisingly hard time getting that to work; the macro forwards the args to an initializer list.
```
    CONSTRUCT_ON_FIRST_USE(std::string, 4194304, 'a');
```

translates into this build error 

```
n file included from source/client/stream_decoder.cc:1:
bazel-out/k8-dbg/bin/source/client/_virtual_includes/nighthawk_client_lib/client/stream_decoder.h:85:41: error: implicit conversion from 'int' to 'const char' changes value from 4194304 to 0 [-Werror,-Wconstant-conversion]
    CONSTRUCT_ON_FIRST_USE(std::string, 4194304, 'a');
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
bazel-out/k8-dbg/bin/external/envoy/source/common/common/_virtual_includes/macros/common/common/macros.h:34:45: note: expanded from macro 'CONSTRUCT_ON_FIRST_USE'
    static const type* objectptr = new type{__VA_ARGS__};                                          \
```

Reply:
Not sure why it's not doing the right thing here, probably some constructor override issue interacting with templating etc. You can always implement the `CONSTRUCT_ON_FIRST_USE` idiom by hand in this file.

Reply:
Done in 1bb39a09507f676f009a2bd62548e9f3daf8db62